### PR TITLE
TW-1360: Add Slidable Chat List Item - Read/Unread

### PR DESCRIPTION
**Pull request by imGok**
https://github.com/linagora/twake-on-matrix/assets/48354990/f42644d7-4f36-4c70-942d-ee7cc3a1dfdd

#1360

### CODE REVIEW

**Comment by sherlockvn** [comment]
- [ ] Please make it synchronize with context menu in chat list for invitation chat list item in chat list
```
List<PopupMenuItem> _popupMenuActionTile(
    BuildContext context,
    Room room,
  ) {
    final listAction = [
      if (room.membership != Membership.invite) ...[
        ChatListSelectionActions.read,
        ChatListSelectionActions.pin,
      ],
      ChatListSelectionActions.mute,
    ];
    return listAction.map((action) {
      return PopupMenuItem(
        padding: EdgeInsets.zero,
        child: popupItemByTwakeAppRouter(
          context,
          action.getTitleContextMenuSelection(context, room),
          iconAction: action.getIconContextMenuSelection(room),
          onCallbackAction: () => _handleClickOnContextMenuItem(
            action,
            room,
          ),
        ),
      );
    }).toList();
  }

```

**Comment by hoangdat** [comment]
Hi we should update the correct color for these actions
image

File: lib/pages/chat_list/chat_list.dart [hunk 1]
```diff
@@ -255,7 +255,7 @@ class ChatListController extends State<ChatList>
     });
   }
 
-  Future<void> toggleUnread() async {
+  Future<void> toggleReadSelections() async {
```
Comment (hoangdat) [inline lib/pages/chat_list/chat_list.dart:258]: all the logic inside relate to `unread`, why the name is `read`?

File: lib/pages/chat_list/chat_list_view_style.dart [hunk 2]
```diff
@@ -11,6 +11,7 @@ class ChatListViewStyle {
       const Size.fromHeight(120);
 
   // Between 0 and 1, scale on actions length
-  static const double slidableExtentRatio = 0.25;
+  static const double slidableExtentRatio = 0.50;
```
Comment (hoangdat) [inline lib/pages/chat_list/chat_list_view_style.dart:14]: why we use the ratio, how about use fixed size? IMO. right now it is too big.

**Comment by sherlockvn** [comment]
PLease fix the below error:
https://github.com/linagora/twake-on-matrix/assets/43041967/c42a5493-f05c-4a33-b5a4-cd6f0324c7e0

File: lib/pages/chat_list/chat_list_view_style.dart (hunk 2 above)
Comment (imGok) [reply lib/pages/chat_list/chat_list_view_style.dart:14]: Sizes are determined like that on slidable lib, extent ratio is the total length of the drawer. I'll check for better solutions

**Comment by imGok** [comment]
Fixed "Unread" cut word
image

File: lib/pages/chat_list/chat_list_view_style.dart [hunk 3]
```diff
@@ -16,6 +16,7 @@ class ChatListViewStyle {
     return slidableSizeRatio * slidablesLength;
   }
 
-  static const Color pinSlidableColorRaw = Color(0xFF00C853);
-  static const Color readSlidableColorRaw = Color(0xFFAEAAAE);
+  static const EdgeInsets slidablePadding = EdgeInsets.all(4.0);
+  static const Color pinSlidableColorRaw = Color(0xFFAEAAAE);
```
Comment (sherlockvn) [inline lib/pages/chat_list/chat_list_view_style.dart:20]: use linagoraSystemColor, because the designer have updated it

Comment (sherlockvn) [reply lib/pages/chat_list/chat_list_view_style.dart:20]: please remember unread/read, pin/unpin have different colors.

Comment (imGok) [reply lib/pages/chat_list/chat_list_view_style.dart]: Hi I fixed colors and icon please check 


https://github.com/linagora/twake-on-matrix/assets/48354990/4706a6a0-385d-437b-910e-69122eb6e77f

**Review by nqhhdev — APPROVED** [review]

**Review by sherlockvn — APPROVED** [review]

**Pull request merged**